src/controller_plugin.py

```diff
@@ -242,6 +243,65 @@ def get_file(self, file_name, encoding=None):
             return content.decode(encoding)
         return content
 
+    def track_job(self, job):
+        """ Tracks a job by location to prevent request_job_for_client from
+            returning new jobs for locations being tracked. Raises a value
+            error if attempting to insert a job that has a location that is
+            already tracked.
+
+        Arguments:
+            job {dict} -- A job for which its location will be tracked until
+                          respond output is called or the tracking is cleared
+        """
+
+        if self.job_location(job) not in self.tracked_jobs:
+            self.tracked_jobs[self.job_location(job)] = job
+        else:
+            raise ValueError("The location: {} is already being tracked."
+            .format(self.job_location(job)))
+
+    def clear_tracking(self, location, is_error=False):
+        """Removes a job from tracking and updates the database
+            To Done or Error
+
+        Arguments:
+            location {str} -- the IP address or hostname to remove
+
+        Keyword Arguments:
+            is_error {bool} -- if the response should indicate an error occured.
+                             (default: {False})
+        """
+
+        if is_error:
+           self.respond_error(
+               self.tracked_jobs[location],
+               "Tracked job at {} has errored".format(location)
+            )
+        else:
+            self.respond_output(
+                self.tracked_jobs[location],
+                "Tracked job at {} has been cleared".format(location)
+            )
+
+        del self.tracked_jobs[location]
+    
+    def untrack(self, location):
+        try:
+            del self.tracked_jobs[location]
+        except KeyError:
+            raise ValueError("No job job found for {}".format(location)) 
```

Comment:
Again with these two, raising an exception might kill the thread unless they are handling it, might not want to do that.